File: src/types/metadata.ts
export interface StoreMetadata {
  gameId: string;
  name: string;
  shortDescription: string | null;
  headerImageUrl: string | null;
  developers: string[];
  publishers: string[];
  genres: GenreInfo[];
  categories: CategoryInfo[];
  screenshots: ScreenshotInfo[];
  releaseDate: string | null;
  metacriticScore: number | null;
  metacriticUrl: string | null;
  steamTags: SteamTagInfo[];
}

export interface GenreInfo {
  id: string;
  description: string;
}

export interface CategoryInfo {
  id: number;
  description: string;
}

export interface ScreenshotInfo {
  id: number;
  thumbnailUrl: string;
  fullUrl: string;
}

export interface SteamTagInfo {
  name: string;
  votes: number;
}

File: src-tauri/src/models/metadata.rs
use serde::{Deserialize, Serialize};

/// Our cached game metadata model, converted from the Store API response.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct StoreMetadata {
    pub game_id: String,
    pub name: String,
    pub short_description: Option<String>,
    pub header_image_url: Option<String>,
    pub developers: Vec<String>,
    pub publishers: Vec<String>,
    pub genres: Vec<GenreInfo>,
    pub categories: Vec<CategoryInfo>,
    pub screenshots: Vec<ScreenshotInfo>,
    pub release_date: Option<String>,
    pub metacritic_score: Option<u32>,
    pub metacritic_url: Option<String>,
    pub steam_tags: Vec<SteamTagInfo>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GenreInfo {
    pub id: String,
    pub description: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CategoryInfo {
    pub id: u32,
    pub description: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ScreenshotInfo {
    pub id: u32,
    pub thumbnail_url: String,
    pub full_url: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SteamTagInfo {
    pub name: String,
    pub votes: u32,
}
